src/hexer/nifcgen.nim

```diff
@@ -498,7 +499,26 @@ proc traverseType(e: var EContext; c: var Cursor; flags: set[TypeFlag] = {}) =
   else:
     error e, "type expected but got: ", c
 
-proc traverseParams(e: var EContext; c: var Cursor) =
+proc maybeByConstRef(e: var EContext; c: var Cursor; pragmas: Cursor) =
+  let param = asLocal(c)
+  if passByConstRef(param.typ, pragmas, sizeof(int)):
+    var paramBuf = createTokenBuf()
+    paramBuf.add tagToken("param", c.info)
+    paramBuf.add param.name
+    paramBuf.add param.exported
+    paramBuf.add param.pragmas
+    copyIntoKind paramBuf, PtrT, param.typ.info:
+      paramBuf.add param.typ
+    paramBuf.add param.val # TODO: default parameters haven't been implemented yet
```

Comment:
There are no default parameters in NIFC!

Reply:
Yeah, it didn't handle default parameters from the frontend because that's implemented yet
```nim
proc foo(x: int = 1) =
  discard

foo()
```
codegen error

Reply:
Just ignore `param.val` here, why pretend that NIFC has it?

Reply:
okay